tensorflow/compiler/mlir/python/mlir.cc

```diff
@@ -249,13 +251,14 @@ std::string ImportGraphDef(const std::string &proto,
   }
 
   // Unmatched input node array and data shapes sizes.
-  if (node_names.size() != node_shapes.size()) {
+  if (node_names_size != node_shapes.size()) {
     auto s = tensorflow::errors::InvalidArgument(
         "Length of input node array and data shape doesn't match");
     Set_TF_Status_from_Status(status, s);
     return "// error";
   }
-  for (unsigned i = 0, e = node_names.size(); i < e; i++) {
+  for (decltype(node_names_size)::size_type i = llvm::seq<decltype(node_names_size)::size_type>(
+          0, node_names.size()) {
```

Comment:
@joker-eph I encourage you to make a new vector type to replace `std::vector` so that its `size_type` will match your philosophy. Otherwise type inconsistencies occur when assigning an `unsigned` when it is actually `signed`.